### DISCUSSION TITLE
Null value inserted differently in Varchar and Decimal64 types

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#258] [Q&A]
I have seen couple of strange behavior of TimeBase while loading data.
Firstly, when null values(from python) were inserted in 'Float Decimal64' data type attribute of stream, it automatically converts it to 0.
But isnt happening same for 'Float Decimal' data type.

Secondly, when we insert null records in Varchar type it stores it as empty string instead of null, hence isnt catching records in "... where column_name is null"  query.

Its getting hard to catch the null entries as 0(decial64) or empty string incase of string means different then null values.

@alex-karpovich  can you please let us know is this an expected behavior of TimeBase?

**Comment by alex-karpovich** [comment]
@ankitsoin 
Nope, this is not expected.
We will push updated version on Monday. (CE version is outdated, so we merge recent changes from EE)

**Reply by ankitsoin** [reply]
Ok Thank you @alex-karpovich

**Comment by Romkisel** [comment]
Hi, @ankitsoin, we've released tbapi python client 6.0.4, which includes fixes in decimals and varchars loading.

**Reply by ankitsoin** [reply]
Thanks @Romkisel and @alex-karpovich for quick action.
One more thing I want to highlight that in QQL, 'in' is not working for decimal64 data types, though it is working for decimal type.
example select * from stream_name where decimal64_col in (22.34, 22.35) 

please let us know if it requires fix as well,

**Reply by alex-karpovich** [reply]
@ankitsoin to work with decimal numbers I suggest to use ranges to avoid rounding errors

```select * from stream_name where decimal64_col >= 22.34 and decimal64_col <= 22.35```

**Reply by ShivaniSK** [reply]
hello @alex-karpovich , so in case we want to exclude the records where decimal64_col values are 1.0, 6.0, 89.0, 55.0 then how should the timebase query look like in that case?

**Reply by alex-karpovich** [reply]
@ShivaniSK I would suggest to use exact matches 
```select * from stream_name where decimal64_col == 1.0 or decimal64_col == 89.0``` etc.
